client/app/lib/helpers/jobHelpers.ts

```diff
@@ -0,0 +1,38 @@
+import { AxiosError } from 'axios';
+import { JobCompleted, JobErrored } from 'types/jobs';
+
+import GlobalAPI from 'api';
+
+const pollJob = (
+  jobUrl: string,
+  onSuccess: (data: JobCompleted) => void,
+  onFailure: (error: JobErrored | AxiosError) => void,
+  pollInterval: number,
+): void => {
+  const poller = setInterval(() => {
+    GlobalAPI.jobs
+      .get(jobUrl)
+      .then((response) => {
+        switch (response.data.status) {
+          case 'submitted':
+            break;
+          case 'completed':
+            clearInterval(poller);
+            onSuccess(response.data);
+            break;
+          case 'errored':
+            clearInterval(poller);
+            onFailure(response.data);
+            break;
+          default:
+            throw new Error('Unknown job status');
```

Comment:
I was going to suggest to also `clearInterval` here, but technically this is an assertion, i.e., ideally, the switch should never, ever come here right?

If yes, then can just ignore this comment.

Reply:
Yup. it's not supposed to go to this branch..